### PR TITLE
[DOCS] Registered domain processor

### PR DESCRIPTION
**Pull request by danhermann**
Documentation for the ingest processor added in #67611.

### CODE REVIEW

File: docs/reference/ingest/processors/registered-domain.asciidoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+[[registered-domain-options]]
+.Registered Domain Options
+[options="header"]
+|======
+| Name               | Required | Default | Description
+| `field`            | yes      |         | Field containing the source FQDN.
+| `target_field`     | no       | `<empty string>` | Field under which the
+extracted domain components will be written. If the empty string, the domain
+components will be added to the root of the document.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/registered-domain.asciidoc:22]: Just some minor rewording here. Feel free to ignore if wanted. It seemed odd to me that `url` wasn't the default to fit ECS, but there's probably a good reason for that.
```suggestion
| `target_field`     | no       | `<empty string>` | Object field containing
extracted domain components. If an `<empty string>`, the processor adds
components to the document's root.
```

File: docs/reference/ingest/processors/registered-domain.asciidoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          "field": "fqdn",
+          "target_field": "url"
+        }
+      }
+    ]
+  },
+  "docs": [
+    {
+      "_source": {
+        "fqdn": "www.amazon.co.uk"
```
Comment (jrodewig) [inline docs/reference/ingest/processors/registered-domain.asciidoc:52]: I'd use `example.com` or `example.org` in any docs or examples.
```suggestion
        "fqdn": "www.example.org"
```

File: docs/reference/ingest/processors/registered-domain.asciidoc [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    {
+      "doc": {
+        ...
+        "_source": {
+          "fqdn": "www.amazon.co.uk",
+          "url": {
+            "subdomain": "www",
+            "registered_domain": "amazon.co.uk",
+            "top_level_domain": "co.uk",
+            "domain": "www.amazon.co.uk"
```
Comment (jrodewig) [inline docs/reference/ingest/processors/registered-domain.asciidoc:74]: Syncs with the `example.org` changes above.
```suggestion
          "fqdn": "www.example.org",
          "url": {
            "subdomain": "www",
            "registered_domain": "example.org",
            "top_level_domain": "org",
            "domain": "www.example.org"
          }
```

File: docs/reference/ingest/processors/registered-domain.asciidoc [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          }
+        }
+      }
+    }
+  ]
+}
+----
+// TESTRESPONSE[s/\.\.\./"_index":"_index","_id":"_id","_ingest":{"timestamp":$body.docs.0.doc._ingest.timestamp},/]
+// NOTCONSOLE
```
Comment (jrodewig) [inline docs/reference/ingest/processors/registered-domain.asciidoc:83]: I tested this locally. We should be able to remove this comment and actually run the test.
```suggestion
```

**Review by jrodewig — APPROVED** [review]
Thanks @danhermann. I left some non-blocking feedback. Feel free to ignore if wanted.

File: docs/reference/ingest/processors/registered-domain.asciidoc (hunk 2 above)
Comment (danhermann) [reply docs/reference/ingest/processors/registered-domain.asciidoc:52]: Good point. I did want something with a two-part TLD to illustrate that it's not just splitting the FQDN on periods. Would something such as `www.example.co.uk` work?

Comment (jrodewig) [reply docs/reference/ingest/processors/registered-domain.asciidoc:52]: I'd use `www.example.ac.uk`. `ac.uk` is reserved for academic use and would be harder to use for nefarious means in the future.

File: docs/reference/ingest/processors/registered-domain.asciidoc (hunk 1 above)
Comment (danhermann) [reply docs/reference/ingest/processors/registered-domain.asciidoc]: > Just some minor rewording here. Feel free to ignore if wanted. It seemed odd to me that `url` wasn't the default to fit ECS, but there's probably a good reason for that.

Fyi, the reason for that is that there are multiple places in ECS where registered domain info may be populated so we didn't want to default to just one of them. See https://github.com/elastic/elasticsearch/pull/67611#discussion_r603368001 for more context.

Comment (jrodewig) [reply docs/reference/ingest/processors/registered-domain.asciidoc]: Makes sense. Thanks for the explanation @danhermann!

**Pull request merged**